### PR TITLE
For #39: Creating test for metadata update

### PR DESCRIPTION
**Pull request by paulodamaso**
Foe #39: created test for metadata update

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#40](https://github.com/artipie/npm-adapter/issues/40) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/177f2ad6-cd38-4ba4-b78b-7184dec7ff8d); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artipie/npm/MetaTest.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import java.util.Optional;
+import javax.json.Json;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.IsEqual;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for {@link Meta}.
+ *
+ * @since 0.1
```
Comment (Vatavuk) [inline src/test/java/com/artipie/npm/MetaTest.java:38]: @paulodamaso since `0.4.2`

**Comment by paulodamaso** [comment]
@Vatavuk Done!

File: src/test/java/com/artipie/npm/MetaTest.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                    (arr1, arr2) ->
+                        ByteBuffer.wrap(
+                            new byte[arr1.length + arr2.length]
+                        ).put(arr1).put(arr2).array()
+                ).blockingGet(),
+                Charset.defaultCharset()
+            ),
+            new IsEqual<>(
+                //@checkstyle LineLengthCheck (1 line)
+                "{\"dist-tags\":{\"latest\":\"1.0.0\",\"alpha\":\"1.0.1\"},\"versions\":{\"1.0.0\":{},\"1.0.1\":{}}}"
```
Comment (g4s8) [inline src/test/java/com/artipie/npm/MetaTest.java:101]: @paulodamaso can we use JSON hamcrest matchers for this test? https://github.com/g4s8/matchers-json

**Comment by 0crat** [comment]
@vatavuk[/z](https://www.0crat.com/u/vatavuk) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by paulodamaso** [comment]
@g4s8 I can't: seems that we have a bug either on on the tested method or in JSON matchers, please take a look at the last commit.

File: src/test/java/com/artipie/npm/MetaTest.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+                    ).put(arr1).put(arr2).array()
+                ).blockingGet()
+            )
+            ).readObject().asJsonObject(),
+
+            new AllOf<>(
+                new ListOf<>(
+                    new JsonHas(
+                        "dist-tags",
+                        new JsonContains(
```
Comment (g4s8) [inline src/test/java/com/artipie/npm/MetaTest.java:112]: @paulodamaso you're trying to match [JSON array](https://github.com/g4s8/matchers-json#matching-json-arrays) with `JsonContains`:
>  Expected: (field "dist-tags" with JSON array with 2 items and field "versions" with JSON array with 2 items)
     but: field "dist-tags" with JSON array with 2 items field "dist-tags" was a org.glassfish.json.JsonObjectBuilderImpl$JsonObjectImpl (<{"latest":"1.0.0","alpha":"1.0.1"}>)

But according to your previous expected string this JSON doesn't have arrays:
```json
{
  "dist-tags": {
    "latest": "1.0.0",
    "alpha": "1.0.1"
  },
  "versions": {
    "1.0.0":{},
    "1.0.1":{}
  }
}
```

So the matcher should be:
```java
new JsonHas(
  "dist-tags",
  Matchers.allOf(
    new JsonHas("latest", new JsonValueIs("1.0.0")),
    new JsonHas("alpha", new JsonValueIs("1.0.1"))
  )
)
```

**Comment by paulodamaso** [comment]
@g4s8 My bad, sorry. But seems that we are having problems asserting the "{}" elements

**Comment by g4s8** [comment]
@paulodamaso you're getting this error because empty string is not the same as empty JSON object, you can compare it using:
```java
new JsonHas("1.0.0", Matchers.any(JsonValue.class))
```

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@Vatavuk please make sure you will find at least 3 issues during next CR, thanks

**Comment by Vatavuk** [comment]
@sereshqua ok

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**